mindmeld/components/entity_resolver.py

```diff
@@ -538,12 +618,101 @@ def _construct_whitelist_query(entity, weight=1, use_phons=False):
 
             return results[0:20]
 
-    def _predict_exact_match(self, entity):
-        """Predicts the resolved value(s) for the given entity using the loaded entity map.
+    def _load(self):
+        """Loads the trained entity resolution model from disk."""
+        try:
+            scoped_index_name = get_scoped_index_name(
+                self._app_namespace, self._es_index_name
+            )
+            if not self._es_client.indices.exists(index=scoped_index_name):
+                self.fit()
+        except EsConnectionError as e:
+            logger.error(
+                "Unable to connect to Elasticsearch: %s details: %s", e.error, e.info
+            )
+            raise EntityResolverConnectionError(es_host=self._es_client.transport.hosts) from e
+        except TransportError as e:
+            logger.error(
+                "Unexpected error occurred when sending requests to Elasticsearch: %s "
+                "Status code: %s details: %s",
+                e.error,
+                e.status_code,
+                e.info,
+            )
+            raise EntityResolverError from e
+        except ElasticsearchException as e:
+            raise EntityResolverError from e
+
+
+class EntityResolverUsingExactMatch(EntityResolverBase):
+    """
+    Resolver class based on exact matching
+    """
+
+    def __init__(self, app_path, resource_loader, entity_type, er_config, **kwargs):
+        super().__init__(app_path, resource_loader, entity_type, er_config, **kwargs)
+        self._normalizer = self.resource_loader.query_factory.normalize
+        self._exact_match_mapping = None
+
+    @staticmethod
+    def _process_entity_map(entity_type, entity_map, normalizer):
+        """Loads in the mapping.json file and stores the synonym mappings in a item_map and a
+        synonym_map for exact match entity resolution when Elasticsearch is unavailable
 
         Args:
-            entity (Entity): An entity found in an input query
+            entity_type: The entity type associated with this entity resolver
+            entity_map: The loaded mapping.json file for the given entity type
+            normalizer: The normalizer to use
+        """
+        item_map = {}
+        syn_map = {}
+        seen_ids = []
+        for item in entity_map.get("entities"):
+            cname = item["cname"]
+            item_id = item.get("id")
+            if cname in item_map:
+                msg = "Canonical name %s specified in %s entity map multiple times"
+                logger.debug(msg, cname, entity_type)
+            if item_id:
```

Comment:
`if item_id  and item_id in seen_ids`

Reply:
With the existing flow, `item_id` can only be added to `seen_ids` if it is non empty. If we combine both the _if_ conditions, we might need to use another _if_ condition for adding. Although I am not sure if `item_id` can be an empty string. I retained this snippet of code as-is from exiting version. Do you think it should be modified, assuming `item_id` is never an empty string?